runners/direct-java/pom.xml

```diff
@@ -361,6 +361,12 @@
       </exclusions>
       <scope>test</scope>
     </dependency>
-
+    
+    <!-- needed for eclipse-jdt generated core as the test-jar references classes from this -->
```

Comment:
runners/direct-java/pom.xml pulls in org.apache.beam:beam-sdks-java-core:test-jar (test scope).   That jar has classes in it that reference dataflow classes.   During the integration tests (not the unit tests), something is scanning the classes or similar that is causing a test failure as the google dataflow classes are not available:

```
testSpecializedButIgnoredGenericInPipeline(org.apache.beam.sdk.coders.CoderRegistryTest)  Time elapsed: 0.012 sec  <<< ERROR!
java.lang.NoClassDefFoundError: com/google/cloud/dataflow/sdk/coders/Proto2CoderTestMessages
	at java.lang.Class.getDeclaringClass0(Native Method)
	at java.lang.Class.getDeclaringClass(Class.java:1235)
	at java.lang.Class.getEnclosingClass(Class.java:1277)
	at java.lang.Class.getSimpleBinaryName(Class.java:1443)
	at java.lang.Class.getSimpleName(Class.java:1309)
	at java.lang.Class.isAnonymousClass(Class.java:1411)
	at org.apache.beam.sdk.transforms.PTransform.getKindString(PTransform.java:243)
	at org.apache.beam.sdk.transforms.PTransform.getName(PTransform.java:201)
	at org.apache.beam.sdk.Pipeline.applyTransform(Pipeline.java:307)
	at org.apache.beam.sdk.values.PCollection.apply(PCollection.java:154)
	at org.apache.beam.sdk.coders.CoderRegistryTest.testSpecializedButIgnoredGenericInPipeline(CoderRegistryTest.java:420)
```
To work around that problem, I added the jar as a test dependency.   It's only the org.apache.beam.sdk.coders.CoderRegistryTest test that is failing so there may be some way to modify that test to avoid the problem as well.   I'm not familiar enough with that yet.